client/connect-account-page/index.js

```diff
@@ -96,7 +100,50 @@ const ConnectPageOnboardingDisabled = () => (
 const ConnectPageOnboarding = () => {
 	const [ isSubmitted, setSubmitted ] = useState( false );
 
-	const handleSetup = () => {
+	const handleLocationCheck = ( availableCountries ) => {
+		// Reset the 'Set up' button state if merchant decided to stop
+		const whenDeclined = () => {
+			setSubmitted( false );
+		};
+		// Redirect the merchant if merchant decided to continue
+		const whenConfirmed = () => {
+			window.location = wcpaySettings.connect.url;
+		};
+
+		// Populate translated list of supported countries we want to render in the modal window.
+		const countries = Object.values( availableCountries )
+			.map( ( country ) => {
+				// eslint-disable-next-line @wordpress/i18n-no-variables
+				return __( country, 'woocommerce-payments' );
+			} )
+			.sort()
+			.map( ( country ) => {
+				return { title: country };
+			} );
+
+		const container = document.createElement( 'div' );
+		container.id = 'wcpay-onboarding-location-check-container';
+		document.body.appendChild( container );
+		ReactDOM.render(
+			<OnboardingLocationCheckModal
+				countries={ countries }
+				whenDeclined={ whenDeclined }
+				whenConfirmed={ whenConfirmed }
+			/>,
+			container
```

Comment:
Nothing wrong with this approach, but any reason for rendering the Modal in this way? The state of opening and closing the modal could be handled with `useState` and rendering of `OnboardingLocationCheckModal` could be controlled by the state and it can be inside the return with other dom elements (around line 167).

Reply:
I adopted this approach from how we handling other modals creation. For this iteration, I'd go with this approach, but I'm curious to explore your approach.

Can you illustrate the approach or perhaps share a link where I can learn from (keep in mind that's my 3rd PR in React :thinking:  ).

Reply:
Please consider this example:
https://github.com/Automattic/woocommerce-payments/blob/7ead98576125cef39ca47229a0aa41254d3eef43/client/overview/inbox-notifications/index.js#L309-L314

This is another way of rendering the modal, controlled by the state.

Reply:
I see, will check if we can apply this.